packages/core/src/running/lintOnce.ts

```diff
@@ -83,30 +84,52 @@ export async function lintOnce(
 		? undefined
 		: await readFromCache(allFilePaths, configDefinition.filePath);
 
-	// TODO: This is very slow and the whole thing should be refactored 🙌.
-	// The separate lintFile function recomputes rule options repeatedly.
-	// It'd be better to group found files together in some way.
-	// Plus, this does an await in a for loop - should it use a queue?
-	for (const filePath of allFilePaths) {
-		const { dependencies, diagnostics, reports } =
-			cached?.get(filePath) ??
-			(await lintFile(
-				makeAbsolute(filePath),
-				languageFactories,
-				useDefinitions
-					.filter((use) => use.found.has(filePath))
-					.flatMap((use) => use.rules),
-				skipDiagnostics,
-			));
-
-		filesResults.set(filePath, {
-			dependencies: new Set(dependencies),
-			diagnostics: diagnostics ?? [],
-			reports: reports ?? [],
-		});
-
-		if (reports?.length) {
-			totalReports += reports.length;
+	const rulesWithOptions = computeRulesWithOptions(
+		useDefinitions.flatMap((use) => use.rules),
+	);
+
+	// TODO: It would probably be good to group rules by language...
+	for (const [rule, options] of rulesWithOptions) {
+		const parsedOptions = await rule.options["~standard"].validate(options);
```

Comment:
There was a casting issue so I went ahead and added this. Happy to revert this out, but I think it doesn't really make sense to do except stacked with these changes because we'd recheck the options otherwise.